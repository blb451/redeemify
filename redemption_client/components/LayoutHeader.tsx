import { StyleSheet } from 'react-native';

import { View, Text } from '@/components/Themed';

import { User } from '@/types';

interface LayoutHeaderProps {
  currentUser: User | null;
  title?: string;
}

export default function LayoutHeader({
  currentUser,
  title,
}: LayoutHeaderProps) {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <View style={styles.pointsContainer}>
        <Text style={styles.points}>Available Points: </Text>
        <Text style={styles.pointValue}>{currentUser?.points}</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignContent: 'center',
    padding: 15,
    paddingTop: 50,
  },
  pointsContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  points: {
    fontWeight: 'bold',
    fontSize: 16,
  },
  pointValue: {
    fontSize: 16,
  },
  title: {
    fontSize: 20,
  },
});
