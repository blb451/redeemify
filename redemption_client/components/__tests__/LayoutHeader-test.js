import { render } from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import LayoutHeader from '@/components/LayoutHeader';

describe('LayoutHeader component', () => {
  const currentUser = {
    id: '1',
    name: 'TestUser',
    points: 100,
  };

  it(`renders snapshot correctly`, () => {
    const tree = renderer
      .create(<LayoutHeader currentUser={currentUser} title="Test Title" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly', () => {
    const { getByText } = render(
      <LayoutHeader currentUser={currentUser} title="Test Title" />
    );

    expect(getByText('Test Title')).toBeTruthy();
    expect(getByText('Available Points:')).toBeTruthy();
    expect(getByText('100')).toBeTruthy();
  });

  test('does not render title if not provided', () => {
    const { queryByText } = render(<LayoutHeader currentUser={currentUser} />);
    expect(queryByText('Test Title')).toBeNull();
  });
});
