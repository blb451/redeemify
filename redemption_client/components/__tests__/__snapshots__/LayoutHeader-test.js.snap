// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutHeader component renders snapshot correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignContent": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 15,
        "paddingTop": 50,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "fontSize": 20,
        },
      ]
    }
  >
    Test Title
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "display": "flex",
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Available Points: 
    </Text>
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "fontSize": 16,
          },
        ]
      }
    >
      100
    </Text>
  </View>
</View>
`;
